### DISCUSSION TITLE
Let eprosima_find_package succeed when target was already configured by top level cmake

### DISCUSSION THREAD

**Discussion by leorich** [eProsima/Fast-DDS#4991] [Support]
Using a top level cmake can be a useful way to build packages together:

```cmake 
cmake_minimum_required(VERSION 3.5)
project(fastdds_top_level LANGUAGES CXX VERSION 2.14.0)

message(STATUS "${PROJECT_NAME} version ${PROJECT_VERSION}")

# build packages
add_subdirectory(Fast-CDR)
add_subdirectory(Fast-DDS)
```

However, on 2.14.0, `eprosima_find_package` would fail on  [line 133](https://github.com/eProsima/Fast-DDS/blob/master/cmake/common/eprosima_libraries.cmake#L133) since it doesn't check to see if the target already exists in the context.

Adding `AND NOT TARGET ${package}` to the condition on [line 131](https://github.com/eProsima/Fast-DDS/blob/master/cmake/common/eprosima_libraries.cmake#L131) would let cmake continue execution.

![image](https://github.com/eProsima/Fast-DDS/assets/30325801/259c8e32-3ec8-46b9-8dc7-706b90702ee0)

**Comment by JesusPoderoso** [comment]
Hi @leorich, thanks for using Fast DDS.
Sorry for the inconvenience. Perhaps the use of Fast DDS [third-party distributed dependencies](https://fast-dds.docs.eprosima.com/en/latest/installation/configuration/cmake_options.html#third-party-libraries-options) may suit your needs.
If you consider contributing to the project, do not hesitate opening a pull request with the suggested fix. Thanks in advance!
